mods/weather/init.lua

```diff
@@ -1,21 +1,12 @@
--- Disable by mapgen or setting
-
+-- Check whether mod should be active
 local mg_name = minetest.get_mapgen_setting("mg_name")
-if minetest.settings:get_bool("enable_weather") == false then
-	return
-end
+local mod_enabled = minetest.settings:get_bool("enable_weather", true)
+local randomize_clouds = mod_enabled and mg_name ~= "v6" and mg_name ~= "singlenode"
 
-if mg_name == "v6" or mg_name == "singlenode" then
-	-- set a default shadow intensity for mgv6 and singlenode
-	minetest.register_on_joinplayer(function(player)
-		player:set_lighting({ shadows = { intensity = 0.33 } })
-	end)
-
-	return
-end
+weather = {}
+local playerlist = {}
```

Comment:
I'd rename this to `playerset` since it conceptually is a set of players who have weather enabled. `*list` definitely is misleading since this is not a list in the Lua sense.